ArcGISRuntimeSDKQt_QMLSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.qml

```diff
@@ -0,0 +1,284 @@
+            id: portal
+            url: "https://www.arcgis.com"
+        }
+
+        PortalItem {
+            id: portalItem
+            portal: portal
+            itemId: "acc027394bc84c2fb04d1ed317aac674"
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                busy.visible = true;
+                offlineMapTask.load();
+            }
+        }
+
+        GraphicsOverlay {
+            id: graphicsOverlay
+            renderer: SimpleRenderer {
+                symbol: SimpleLineSymbol {
+                    style: Enums.SimpleLineSymbolStyleSolid
+                    color: "red"
+                    width: 5
+                }
+            }
+        }
+
+        OfflineMapTask {
+            id: offlineMapTask
+            onlineMap: onlineMap
+
+            onErrorChanged: console.log(error.message, error.additionalMessage);
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                offlineMapTask.preplannedMapAreas();
+                path = outputMapPackage;
+                fileFolder.url = path;
+                let worked = fileFolder.makeFolder();
+                if (!worked)
+                    console.log("Make Folder Failed");
+            }
+
+            onPreplannedMapAreasStatusChanged: {
+                if (preplannedMapAreasStatus !== Enums.TaskStatusCompleted)
+                    return;
+
+                preplannedCombo.model = offlineMapTask.preplannedMapAreaList;
+                busy.visible = false;
+
+                for (let i = 0; i < offlineMapTask.preplannedMapAreaList.count; i++) {
+                    offlineMapTask.preplannedMapAreaList.get(i).loadStatusChanged.connect(function () {
+                        if (offlineMapTask.preplannedMapAreaList.get(i).loadStatus !== Enums.LoadStatusLoaded)
+                            return;
+
+                        if (i === 0)
+                            checkIfFileExists(i);
+
+                        var graphic = ArcGISRuntimeEnvironment.createObject("Graphic", { geometry: offlineMapTask.preplannedMapAreaList.get(i).areaOfInterest });
+
+                        graphicsOverlay.graphics.append(graphic);
+                    });
+                    offlineMapTask.preplannedMapAreaList.get(i).load();
+                }
+            }
+
+            onCreateDefaultDownloadPreplannedOfflineMapParametersStatusChanged: {
+                if (createDefaultDownloadPreplannedOfflineMapParametersStatus !== Enums.TaskStatusCompleted)
+                    return;
+
+                createDefaultDownloadPreplannedOfflineMapParametersResult.updateMode = Enums.PreplannedUpdateModeNoUpdates;
+                let result = createDefaultDownloadPreplannedOfflineMapParametersResult;
+
+                path = outputMapPackage +"/" + result.preplannedMapArea.portalItem.title + ".mmpk";
```

Comment:
this should not get an `.mmpk` extension because it will download to a directory rather than an archived file. We have the same problem in our current `GenerateOfflineMap` sample I think. Might be good to clean that up at the same time?